include/dlaf/lapack/tile.h

```diff
@@ -232,18 +209,40 @@ class CusolverPotrfInfo {
 }
 
 template <class T>
-internal::CusolverPotrfInfo<T> potrfInfo(cusolverDnHandle_t handle, const blas::Uplo uplo,
-                                         const matrix::Tile<T, Device::GPU>& a) {
+internal::CusolverInfo<T> hegst(cusolverDnHandle_t handle, const int itype, const blas::Uplo uplo,
+                                const matrix::Tile<T, Device::GPU>& a,
+                                const matrix::Tile<T, Device::GPU>& b) {
+  DLAF_ASSERT(square_size(a), a);
+  DLAF_ASSERT(square_size(b), b);
+  DLAF_ASSERT(a.size() == b.size(), a, b);
+  const int n = a.size().rows();
+
+  int workspace_size;
+  internal::CusolverHegst<T>::callBufferSize(handle, itype, util::blasToCublas(uplo), n,
+                                             util::blasToCublasCast(a.ptr()), a.ld(),
+                                             util::blasToCublasCast(b.ptr()), b.ld(), &workspace_size);
+  internal::CusolverInfo<T> info{std::max(1, workspace_size)};
+  internal::CusolverHegst<T>::call(handle, itype, util::blasToCublas(uplo), n,
+                                   util::blasToCublasCast(a.ptr()), a.ld(),
+                                   util::blasToCublasCast(b.ptr()), b.ld(),
+                                   util::blasToCublasCast(info.workspace()), info.info());
+
+  return info;
+}
```

Comment:
Is this `hegstInfo`? Or is hegst different from potrf in some way?

Reply:
This comment probably belongs to #379...

Reply:
Please check new commits on #379.